plugins/linux-pipewire/pipewire.c

```diff
@@ -719,27 +719,33 @@ static void on_process_cb(void *user_data)
 			bitmap = SPA_MEMBER(cursor, cursor->bitmap_offset,
 					    struct spa_meta_bitmap);
 
-		if (bitmap && bitmap->size.width > 0 &&
-		    bitmap->size.height > 0 &&
-		    lookup_format_info_from_spa_format(
-			    bitmap->format, NULL, &gs_format, &swap_red_blue)) {
-			const uint8_t *bitmap_data;
-
-			bitmap_data =
-				SPA_MEMBER(bitmap, bitmap->offset, uint8_t);
-			obs_pw->cursor.hotspot_x = cursor->hotspot.x;
-			obs_pw->cursor.hotspot_y = cursor->hotspot.y;
-			obs_pw->cursor.width = bitmap->size.width;
-			obs_pw->cursor.height = bitmap->size.height;
-
+		if (bitmap) {
 			g_clear_pointer(&obs_pw->cursor.texture,
 					gs_texture_destroy);
```

Comment:
Actually, please let me suggest a stylistic change. Looking at the file itself, instead of the diff, seems like there's too much nesting happening. So I think I'd prefer if this was written as:

```c
if (bitmap)
	g_clear_pointer(&obs_pw->cursor.texture, gs_texture_destroy);

if (bitmap && width > 0 && height > 0 && [...]) {
	[all of the current code]

	assert(obs_pw->cursor.texture == NULL);
	obs_pw->cursor.texture = ...;
}
```

What do you think?

Reply:
Sure, I don't mind. Certainly makes the diff more readable, done.